### DISCUSSION TITLE
issue:  Can't reset Controller type = "file"

### DISCUSSION THREAD

**Discussion by CaioDev19** [react-hook-form/react-hook-form#9793] [Q&A]
### Version Number

7.42.1

### Codesandbox/Expo snack

https://github.com/CaioDev19/NewsBlog

### Steps to reproduce

I have the following reusable Controlled Input - path: ./src/components/form/Input

```
interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
  name: string
  control: Control<any, any>
  className?: string
  handleChange?: (e: ChangeEvent<HTMLInputElement>) => void
}

export function Input({
  name,
  placeholder,
  type,
  control,
  className,
  handleChange,
  ...rest
}: Props) {
  return (
    <Controller
      name={name}
      control={control}
      render={({ field: { onChange, value, ref, name } }) => {
        return (
          <Sc.StyledInput
            className={className}
            type={type}
            placeholder={placeholder}
            name={name}
            onChange={
              type === "file"
                ? (e) => {
                    if (typeof handleChange !== "undefined") {
                      handleChange(e)
                    }
                  }
                : onChange
            }
            value={type === "file" ? undefined : value}
            ref={ref}
            {...rest}
          />
        )
      }}
    />
  )
}
```

The handle change function sets the value of the field - path: ./src/pages/Adm/CreatePost
```
handleChange={(e: ChangeEvent<HTMLInputElement>) => {
                  if (!e.target.files) {
                    return
                  }
                  setValue("image", e.target.files)
} 
```

Default Values: 
```
defaultValues: {
      title: "",
      sinopse: "",
      category: "",
      image: null as FileList | null,
}
```

### Expected behaviour

Whenever i try to reset the image field the input still remains with the fileName, i've tried to use setValue, resetField and reset and no one is fully resetting the input. 
Side note the value itself is emptied though the input field is still with the filename.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
1. Building a controlled component which supports file, you can use a third-party lib
2. connect/wrap it with `Controller` or `useController`

**Reply by CaioDev19** [reply]
Is there a way to do it without a third-part lib?

**Comment by CaioDev19** [comment]
Update, the only solution a found do this problem was to make the file input a uncontrolled field and adding a custom onChange as following: 
```
const imageRegister = register("image")
<input
  type="file"
  {...imageRegister}
  onChange={(e) => {
    imageRegister.onChange(e)
    addImage(e.target.files)
  }}
/>
```
I don't now if this is the correct way to do it but was the only one i found.
